apps/evm_family/evm_user_verification.c

```diff
@@ -0,0 +1,223 @@
+
+/*****************************************************************************
+ * PRIVATE TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * STATIC FUNCTIONS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTIONS
+ *****************************************************************************/
+
+bool evm_verify_eth_transfer(const evm_txn_context_t *txn_context) {
+  bool status = false;
+  char address[43] = "0x";
+  const uint8_t *to_address = NULL;
+  const char *unit = NULL;
+  char value[34] = {'\0'};
+  char hex_str[30] = {'\0'};
+  char display[40] = "";
+  const char *title = eth_get_address_title(&txn_context->transaction_info);
+
+  // TODO: verify transaction nonce; this is pending on settings option
+#if 0
+  // TODO: convert byte to value
+  if (!core_scroll_page("Verify nonce", address, evm_send_error)) {
+    return status;
+  }
+#endif
+
+  // verify recipient address; TODO: handle harmony address encoding
+  eth_get_to_address(&txn_context->transaction_info, &to_address);
+  ethereum_address_checksum(
+      to_address, &address[2], false, g_evm_app->chain_id);
+  if (!core_scroll_page(title, address, evm_send_error)) {
+    return status;
+  }
+
+  // verify recipient amount
+  uint8_t len = eth_get_value(&txn_context->transaction_info, hex_str);
+  if (!convert_byte_array_to_decimal_string(
+          len, evm_get_decimal(txn_context), hex_str, value, sizeof(value))) {
+    evm_send_error(ERROR_COMMON_ERROR_UNKNOWN_ERROR_TAG, 1);
+    return status;
+  }
+
+  unit = evm_get_asset_symbol(txn_context);
+  snprintf(display, sizeof(display), UI_TEXT_VERIFY_AMOUNT, value, unit);
+  if (!core_confirmation(display, evm_send_error)) {
+    return status;
+  }
+
+  // verify transaction fee
+  unit = g_evm_app->lunit_name;
+  eth_get_fee_string(
+      &txn_context->transaction_info, value, sizeof(value), ETH_DECIMAL);
+  snprintf(display, sizeof(display), UI_TEXT_SEND_TXN_FEE, value, unit);
+  if (!core_scroll_page(UI_TEXT_TXN_FEE, display, evm_send_error)) {
+    return status;
+  }
+
+  return true;
+}
+
+bool evm_verify_clear_signing(const evm_txn_context_t *txn_context) {
+  bool status = false;
```

Comment:
We don't really need status variable as every return is deterministic here

Reply:
removed